### PR TITLE
Add `onKeyEvent` to `useFocusNode`

### PR DESCRIPTION
**Pull request by kdelorey**
# Summary
The `FocusNode` object has a `onKeyEvent` that is meant to replace the `onKey` callback which uses the legacy `RawKeyEvent` class.  This pull request simply adds the ability to also specify `onKeyEvent` with `useFocusNode()`

```dart
final focusNode = useFocusNode(onKeyEvent: (node, event) {
    // Do something with the keyboard event here
    return KeyEventResult.ignored;
});
```

### CODE REVIEW

File: packages/flutter_hooks/test/use_focus_node_test.dart [hunk 1]
```diff
@@ -128,14 +139,17 @@ void main() {
...
-        );
+            debugLabel: 'Bar', onKey: onKey2, onKeyEvent: onKeyEvent2);
         return Container();
       }),
     );
 
     expect(focusNode.onKey, onKey, reason: 'onKey has no setter');
+    expect(
+      focusNode.onKeyEvent,
+      onKeyEvent,
```
Comment (rrousselGit) [inline packages/flutter_hooks/test/use_focus_node_test.dart:150]: That likely should be onKeyEvent2. The event handlers are mutable now, so hooks should update them

File: packages/flutter_hooks/test/use_focus_node_test.dart [hunk 2]
```diff
@@ -128,14 +139,17 @@ void main() {
     await tester.pumpWidget(
       HookBuilder(builder: (_) {
         focusNode = useFocusNode(
-          debugLabel: 'Bar',
-          onKey: onKey2,
-        );
+            debugLabel: 'Bar', onKey: onKey2, onKeyEvent: onKeyEvent2);
```
Comment (rrousselGit) [inline packages/flutter_hooks/test/use_focus_node_test.dart:142]:
```suggestion
            debugLabel: 'Bar', onKey: onKey2, onKeyEvent: onKeyEvent2,);
```

and format

**Comment by rrousselGit** [comment]
Thanks!

A few smalls changes and it should be good to go.

**Review by rrousselGit — APPROVED** [review]

File: packages/flutter_hooks/test/use_focus_node_test.dart (hunk 1 above)
Comment (kdelorey) [reply packages/flutter_hooks/test/use_focus_node_test.dart]: I updated both onKey and onKeyEvent to support the ability to mutate them.

**Comment by kdelorey** [comment]
@rrousselGit I think I've got your requested changes in, please recheck.

**Comment by rrousselGit** [comment]
Looking good, thanks for the PR!

**Pull request merged**